src/pcre2_auto_possess.c

```diff
@@ -479,13 +480,13 @@ switch(c)
 
   case OP_NCLASS:
   case OP_CLASS:
-#ifdef SUPPORT_WIDE_CHARS
```

Comment:
I would add some `// revert` comments because we should revert these after ECLASS will have a single bitset. It is easy to forget them otherwise.

Reply:
I'm not quite sure what you intend here. Even when ECLASS has a single bitset, it will behave roughly the same as XCLASS, for the purposes of the auto-possessification code at least. There will always need to be the LINK_SIZE slot, holding the size of the stack which follows.

I'll add the TODO comment as you request.

Reply:
Wide chars mean unicode or > 8 bit. In 8 bit mode, without utf, the regular OP_CLASS with 256 bits must handle the extended class, so it should never encounter with this opcode. In general, eclasses should be turned to normal classes/nclasses whenever it is possible, just like xclasses.

Reply:
As for the implementation, eclasses should have a prediction system, which predicts their size if they cannot be turned to class/nclass. The prediction is discarded, if they can. The prediction bit (can or cannot) can be saved in the META code (you have 16 free bits for that) during the byte code length computation phase, and reused when the actual byte code is generated.

Reply:
Of course, I see! If the constant-folding is sufficiently aggressive, then OP_ECLASS will never be emitted without wide-char support, the same as XCLASS. OK, that makes good sense.

Reply:
Btw, both the "need a bitset for eclass" and "eclass can be turned to class/nclass" bits can be computed when the byte code length is computed (two bits in the META code), and this simplifies the actual byte code generation later, since "the oracle" tell us what to do.